---
title: "2025: research in review - Nature Biotechnology"
source: "https://www.nature.com/articles/s41587-025-02961-w"
published: 2025-12-04
created: 2025-12-16
description: "Nature Biotechnology editors pick their favorite research articles from 2025."
tags:
  - "clippings"
---
*Nature Biotechnology* editors pick their favorite research articles from 2025.

This year, biotechnology was characterized by accelerated convergence between gene editing, cell and gene therapy, immunotherapy, and artificial intelligence (AI). CRISPR-based editing tools are more precise and clinically relevant, and biotech continues to push the boundaries of immunotherapy. Quantum computing has also made it into our list of favorite articles published this past year.

## In vivo generated cellular therapies

In vivo strategies for chimeric antigen receptor T cell therapies engineer a patient’s own T cells, without the need for complex external manufacturing processes and chemotherapies. A notable paper this year used targeted lipid nanoparticles for mRNA delivery to specific T cell subsets for in vivo reprogramming [^1]. Biotech companies are also shifting their focus towards in vivo-generated chimeric antigen receptor approaches [^2].

## Recombinases and retrotransposons for large-scale genome engineering

Just when you thought there were enough genome-editing systems, retrotransposons and bridge recombinases appeared and made a splash. A rationally engineered version of a bridge recombinase was shown to efficiently insert, excise and invert large regions of DNA [^3], and the fusion of a retrotransposon to a CRISPR–Cas9 nickase could scarlessly insert a single base or as much as 12.7 kilobases into the genome [^4].

## In vivo base editing for rare disease

A customized lipid-nanoparticle-delivered base-editing therapy was developed for KJ, an infant born with a rare metabolic disease known as CPS1 deficiency [^5]. The treatment was safely administered, providing a way forward for other individuals with rare diseases for whom no medical treatments are available. Although the methodology still needs to be scaled to fit other patients and needs, it is an amazing achievement that brings editing technology into clinical practice.

## Computational metabolomics

Tandem mass spectroscopy is used to characterize biological and environmental samples at the molecular level, but it has always been challenging to interpret the resulting mass spectra, and only 2% of molecular structures can usually be annotated in an untargeted metabolomics experiment using existing reference libraries. A new transformer model was built on a high-quality dataset and the resulting method and atlas can now be used for a wide variety of annotation tasks [^6].

## Quantum computing for drug discovery

Quantum generative models were recently combined with more classical drug-discovery approaches to generate compounds to target the KRAS protein [^7]. KRAS is a very complex protein, so it has been historically difficult to discover a drug that acts against it. The classical-quantum workflow sampled 1 million compounds and then screened them for pharmacological viability, testing 15 candidates.

## Agentic automation for laboratory experiments

Agentic AI systems can now use large language models to streamline workflows, reduce human efforts and accelerate the pace of research in the laboratory. Although AI co-scientists have been around for several years, this year they helped users to select CRISPR systems [^8] and find new gene transfer pathways [^9]. With new resources such as [ToolUniverse](https://github.com/mims-harvard/ToolUniverse), more scientists will be using them to help to guide their own work in the future.

## Moving from animal models to new preclinical systems

Earlier this year, the US Food and Drug Administration announced a plan to phase out animal testing for antibodies and drugs. Tumor-on-a-chip models are a promising alternative; for example, a recent study showed that they are complementary to existing preclinical models for studying immunotherapy response, and show patterns of dynamic immune cell recruitment into the tumor microenvironment [^10]. Organ-on-chip technologies can also benefit low- and middle-income countries, as they require less infrastructure than animal facilities [^11].

## Personalized cancer vaccines here to stay

Two studies published this year showed that vaccines tailored towards specific tumor mutations in early clinical trials were successful, demonstrating the potential of personalized vaccines to change the course of certain cancer types. The first targeted renal cell carcinoma in nine patients [^12]; the second, eight patients with pancreatic ductal adenocarcinoma [^13].

## Precision microbiome programming

Using engineered microorganisms as therapeutics has failed previously in the clinic because there are difficulties with long-term engraftment in the gut. A paper published earlier this year overcomes this challenge — a strain of *Bacteroides vulgatus* was engineered to include a gene module designed to break down oxalate, a compound that forms kidney stones, and treat enteric hyperoxaluria [^14]. The bacteria were shown to colonize without drastically changing the diversity of the native microbiota.

## An effective recombinant antivenom finally realized

Developing broad-spectrum antivenoms has been challenging for biotech, for many reasons [^15]. A nanobody-based recombinant antivenom was shown this year to neutralize 17 of the 18 most medically relevant elapid snakes in sub-Saharan Africa, using only 8 heavy-chain antibodies in the mixture [^16].

These are only a subset of the articles that the editors of *Nature Biotechnology* read and enjoyed this year. Please also see the list of our [top ten](https://doi.org/10.1038/s41587-025-02954-9) news stories, also published in this issue.

## References

## Rights and permissions

[^1]: Hunter, T. L. et al. *Science* **388**, 1311–1317 (2025).

[^2]: Bot, A. et al. *Nat. Rev. Drug Discov.*[https://doi.org/10.1038/s41573-025-01291-5](https://doi.org/10.1038/s41573-025-01291-5) (2025).

[^3]: Perry, N. T. et al. *Science* [https://doi.org/10.1126/science.adz0276](https://doi.org/10.1126/science.adz0276) (2025).

[^4]: Fell, C. W. et al. *Nature* **642**, 1080–1089 (2025).

[^5]: Musunuru, K. et al. *N. Engl. J. Med.***392**, 2235–2243 (2025).

[^6]: Bushuiev, R. et al. *Nat. Biotechnol.*[https://doi.org/10.1038/s41587-025-02663-3](https://doi.org/10.1038/s41587-025-02663-3) (2025).

[^7]: Vakili, M. G. et al. *Nat. Biotechnol.*[https://doi.org/10.1038/s41587-024-02526-3](https://doi.org/10.1038/s41587-024-02526-3) (2025).

[^8]: Qu, Y. et al. *Nat. Biomed. Eng.*[https://doi.org/10.1038/s41551-025-01463-z](https://doi.org/10.1038/s41551-025-01463-z) (2025).

[^9]: Penadés, J. R. et al. *Cell* **188**, 6654–6665 (2025).

[^10]: Liu, H. et al. *Nat. Biotechnol.*[https://doi.org/10.1038/s41587-025-02845-z](https://doi.org/10.1038/s41587-025-02845-z) (2025).

[^11]: Del Valle, A. C. *Nature* **644**, 338 (2025).

[^12]: Braun, D. A. et al. *Nature* **639**, 474–482 (2025).

[^13]: Sethna, Z. et al. *Nature* **639**, 1042–1051 (2025).

[^14]: Whitaker, W. R. et al. *Science* **389**, 303–308 (2025).

[^15]: *Nat. Biotechnol*. **42**, 537–538 (2024).

[^16]: Ahmadi, S. et al. *Nature* **647**, 716–725 (2025).