---
url: https://t.co/wtCW3beYlv
title: "An expanded registry of candidate cis-regulatory elements | Nature"
clipped: 2026-01-08 10:43
source: browser-history
---

# An expanded registry of candidate cis-regulatory elements | Nature

> Source: [https://t.co/wtCW3beYlv](https://t.co/wtCW3beYlv)

An expanded registry of candidate *cis*-regulatory elements

[Download PDF](/articles/s41586-025-09909-9.pdf)

[Download PDF](/articles/s41586-025-09909-9.pdf)

### Subjects

* [Data integration](/subjects/data-integration)
* [Epigenomics](/subjects/epigenomics)
* [Gene regulation](/subjects/gene-regulation)
* [Genetic databases](/subjects/genetic-databases)
* [Transcriptional regulatory elements](/subjects/transcriptional-regulatory-elements)

## Abstract

Mammalian genomes contain millions of regulatory elements that control the complex patterns of gene expression[1](/articles/s41586-025-09909-9#ref-CR1 "Wittkopp, P. J. & Kalay, G. cis-Regulatory elements: molecular mechanisms and evolutionary processes underlying divergence. Nat. Rev. Genet. 13, 59–69 (2011)."). Previously, the ENCODE consortium mapped biochemical signals across hundreds of cell types and tissues and integrated these data to develop a registry containing 0.9 million human and 300,000 mouse candidate *cis*-regulatory elements (cCREs) annotated with potential functions[2](/articles/s41586-025-09909-9#ref-CR2 "ENCODE Project Consortium. Perspectives on ENCODE. Nature 583, 693–698 (2020)."). Here we have expanded the registry to include 2.37 million human and 967,000 mouse cCREs, leveraging new ENCODE datasets and enhanced computational methods. This expanded registry covers hundreds of unique cell and tissue types, providing a comprehensive understanding of gene regulation. Functional characterization data from assays such as STARR-seq[3](/articles/s41586-025-09909-9#ref-CR3 "Arnold, C. D. et al. Genome-wide quantitative enhancer activity maps identified by STARR-seq. Science 339, 1074–1077 (2013)."), massively parallel reporter assay[4](/articles/s41586-025-09909-9#ref-CR4 "Agarwal, V. Massively parallel characterization of transcriptional regulatory elements. Nature 639, 411–420 (2025)."), CRISPR perturbation[5](/articles/s41586-025-09909-9#ref-CR5 "Reilly, S. K. et al. Direct characterization of cis-regulatory elements and functional dissection of complex genetic associations using HCR-FlowFISH. Nat. Genet. 53, 1166–1176 (2021)."),[6](/articles/s41586-025-09909-9#ref-CR6 "Yao, D. et al. Multicenter integrated analysis of noncoding CRISPRi screens. Nat. Methods 21, 723–734 (2024).") and transgenic mouse assays[7](/articles/s41586-025-09909-9#ref-CR7 "Visel, A., Minovitsky, S., Dubchak, I. & Pennacchio, L. A. VISTA enhancer browser—a database of tissue-specific human enhancers. Nucleic Acids Res. 35, D88–D92 (2007).") have profiled more than 90% of human cCREs, revealing complex regulatory functions. We identified thousands of novel silencer cCREs and demonstrated their dual enhancer and silencer roles in different cellular contexts. Integrating the registry with other ENCODE annotations facilitates genetic variation interpretation and trait-associated gene identification, exemplified by the identification of *KLF1* as a novel causal gene for red blood cell traits. This expanded registry is a valuable resource for studying the regulatory genome and its impact on health and disease.

### Similar content being viewed by others

![](https://media.springernature.com/w215h120/springer-static/image/art%3A10.1038%2Fs41586-024-08430-9/MediaObjects/41586_2024_8430_Fig1_HTML.png)

### [Massively parallel characterization of transcriptional regulatory elements](https://www.nature.com/articles/s41586-024-08430-9?fromPaywallRec=false)

Article
Open access
15 January 2025

![](https://media.springernature.com/w215h120/springer-static/image/art%3A10.1038%2Fs41588-021-00900-4/MediaObjects/41588_2021_900_Fig1_HTML.png)

### [Direct characterization of *cis*-regulatory elements and functional dissection of complex genetic associations using HCR–FlowFISH](https://www.nature.com/articles/s41588-021-00900-4?fromPaywallRec=false)

Article
29 July 2021

![](https://media.springernature.com/w215h120/springer-static/image/art%3A10.1038%2Fs41467-024-51921-6/MediaObjects/41467_2024_51921_Fig1_HTML.png)

### [Regulatory transposable elements in the encyclopedia of DNA elements](https://www.nature.com/articles/s41467-024-51921-6?fromPaywallRec=false)

Article
Open access
31 August 2024

## Main

Mammalian genomes are extensive repositories of DNA-encoded instructions that control cellular functions through complex regulatory mechanisms. Central to this regulation are *cis*-regulatory elements (CREs)—non-coding DNA sequences that control the transcription of nearby genes[1](/articles/s41586-025-09909-9#ref-CR1 "Wittkopp, P. J. & Kalay, G. cis-Regulatory elements: molecular mechanisms and evolutionary processes underlying divergence. Nat. Rev. Genet. 13, 59–69 (2011)."). Usually associated with open chromatin and specific histone modifications, CREs contain binding sites for transcription factors and other chromatin-associated proteins that interact with one another and the transcriptional machinery to regulate gene expression. Understanding the biological contexts and functions of CREs is essential for deciphering genome function and its impact on human health and disease[8](/articles/s41586-025-09909-9#ref-CR8 "Lee, T. I. & Young, R. A. Transcriptional regulation and its misregulation in disease. Cell 152, 1237–1251 (2013).").

The ENCODE project has made major contributions towards our understanding of gene regulation by systematically identifying and annotating functional elements in human and mouse genomes[2](/articles/s41586-025-09909-9#ref-CR2 "ENCODE Project Consortium. Perspectives on ENCODE. Nature 583, 693–698 (2020)."). Across tens of thousands of high-throughput functional genomics assays, the ENCODE consortium has comprehensively mapped biochemical signatures and used them to annotate CREs, including the ENCODE Phase III (ENCODE3) registry of cCREs[9](/articles/s41586-025-09909-9#ref-CR9 "ENCODE Project Consortium et al. Expanded encyclopaedias of DNA elements in the human and mouse genomes. Nature 583, 699–710 (2020).").

Here we have expanded the registry of cCREs to include 2.37 million human and 967,000 mouse elements. This expansion leverages new datasets produced during ENCODE Phase IV (ENCODE4) and improved computational methods, making the registry one of the most extensive repositories of CREs available. The updated registry spans hundreds of unique cell and tissue types, and enhances our understanding of gene regulation across a broad range of biological contexts.

In addition to expanding the number of annotated elements, the updated registry integrates functional characterization data for more than 97% of human cCREs, revealing how sequence features influence regulatory activity and uncover new functional subclasses of cCREs. For example, we identified thousands of silencer cCREs—many of which act as enhancers in alternative cellular contexts—and defined MAFF and MAFK-bound cCREs as dynamic enhancers poised for activation under stress-responsive conditions. Together, these analyses highlight the diversity and context-dependent versatility of CRE function.

Integrating the expanded registry with other ENCODE Encyclopedia annotations further enables the systematic interpretation of genetic variation and the identification of trait-associated genes. As an example, we used the registry to nominate *KLF1* as a probable causal gene for red blood cell (RBC) traits. Collectively, this work establishes an expanded and functionally characterized registry that provides a comprehensive framework for studying CREs and their roles in gene regulation, cellular identity and disease.

## Expansion of the ENCODE registry

The ENCODE4 registry encompasses 2,373,014 human cCREs and 926,843 mouse cCREs, covering 21% of the human and 9% of the mouse genomes (Fig. [1](/articles/s41586-025-09909-9#Fig1) and Supplementary Note [1](/articles/s41586-025-09909-9#MOESM1)). This is a threefold increase over ENCODE3 and establishes the ENCODE4 registry as one of the most extensive repositories of CREs currently available (Supplementary Note [1](/articles/s41586-025-09909-9#MOESM1) and Supplementary Data [1](/articles/s41586-025-09909-9#MOESM4) and [2](/articles/s41586-025-09909-9#MOESM4)).

**Fig. 1: The updated registry of cCREs.**

[![figure 1](//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41586-025-09909-9/MediaObjects/41586_2025_9909_Fig1_HTML.png)](/articles/s41586-025-09909-9/figures/1)

**a**,**b**, Schematic of the pipeline used to make version 4 of the registry of cCREs (**a**). We defined element anchors by generating rDHSs and representative transcription factor clusters (TF rClusters). Element anchors are scored and classified with H3K4me3, H3K27ac and CTCF ChIP–seq and ATAC-seq signals (yellow box) according to the scheme in **b**. This results in 2.37 million cCREs in the human genome and 967,000 in the mouse genome. We supplement the registry with additional ENCODE Encyclopedia annotations including transcription quantifications, 3D chromatin contacts, functional characterization measurements, sequence features and genetic variation (blue box). The registry of cCREs and all layered annotations are housed in our web portal SCREEN. New components of the pipeline are denoted by stars. **b**, Overview of our cCRE classification scheme. cCREs are classified on the basis of their patterns of biochemical signals (chromatin accessibility in green, H3K4me3 in red, H3K27ac in yellow, CTCF in blue and transcription factor in purple) and distance from annotated TSSs. High signals are denoted by peaks. ‘+/−’ indicates that the corresponding signal may or may not be present and its presence does not affect classification. New categories of elements are denoted by stars. **c**, Bar graphs depicting the number of cCREs annotated in each class for human and mouse.

[Full size image](/articles/s41586-025-09909-9/figures/1)

This expansion reflects both incorporation of new ENCODE4 data and technical improvements to the cCRE pipeline (Supplementary Note [1](/articles/s41586-025-09909-9#MOESM1) and Supplementary Table [1](/articles/s41586-025-09909-9#MOESM3)). We continued to anchor cCREs on representative DNase hypersensitivity sites (rDHSs) and annotated their functions using H3K4me3, H3K27ac and CTCF chromatin immunoprecipitation with sequencing (ChIP–seq) data. In this expansion, we also incorporated thousands of transcription factor ChIP–seq and assay for transposase-accessible chromatin using sequencing (ATAC-seq) datasets, enabling annotation of elements in low-accessibility regions and biosamples lacking DNase-seq (Supplementary Fig. [1](/articles/s41586-025-09909-9#MOESM1) and Supplementary Tables [2](/articles/s41586-025-09909-9#MOESM3) and [3](/articles/s41586-025-09909-9#MOESM3)). Finally, pipeline updates improved recovery of cCREs in previously difficult-to-map regions, including duplicated loci and repetitive elements such as *Alu* elements (Supplementary Note [1.2](/articles/s41586-025-09909-9#MOESM1) and Supplementary Fig. [2](/articles/s41586-025-09909-9#MOESM1)).

The number of human chromatin accessibility and ChIP–seq experiments increased by 2.3-fold between ENCODE3 and ENCODE4. This doubled the number of biosamples—unique tissues, cell types and cellular states—represented in the registry from 839 to 1,679, spanning 42 human organs and tissues (Extended Data Fig. [1a](/articles/s41586-025-09909-9#Fig5)). Most biosamples are primary cells and tissues, but they also include in vitro-differentiated cells, organoids and cell lines (Extended Data Fig. [1b](/articles/s41586-025-09909-9#Fig5)), supporting both mechanistic studies and translational models. Coordinated ENCODE4 data production also expanded the core collection of biosamples—those with all four core assays DNase, H3K4me3, H3K27ac and CTCF—nearly sevenfold, from 25 in ENCODE3 to 170 in ENCODE4 (Extended Data Fig. [1c](/articles/s41586-025-09909-9#Fig5)), enabling thorough annotation across cell types (Supplementary Note [1.3](/articles/s41586-025-09909-9#MOESM1)).

## Classification of cCREs

We classified cCREs into eight categories based on distance to annotated transcription start sites (TSSs) and combinations of biochemical signals (Fig. [1b,c](/articles/s41586-025-09909-9#Fig1), Supplementary Note [1.4](/articles/s41586-025-09909-9#MOESM1) and Supplementary Fig. [3](/articles/s41586-025-09909-9#MOESM1)). Five of these classes—promoter, proximal enhancer, distal enhancer, CA-H3K4me3 and CA-CTCF—were defined in our previous work[9](/articles/s41586-025-09909-9#ref-CR9 "ENCODE Project Consortium et al. Expanded encyclopaedias of DNA elements in the human and mouse genomes. Nature 583, 699–710 (2020)."). Here we add three new categories: CA-TF cCREs, which have high accessibility but lack H3K4me3, H3K27ac or CTCF and are bound by a transcription factor; CA cCREs, which are accessible but lack enrichment for H3K4me3, H3K27ac and CTCF; and TF cCREs, which show little accessibility or histone marks yet are bound by a transcription factor. These new categories do not by themselves define discrete functional classes; however, subsequent analyses show that they include silencers and dynamic enhancers, extending the registry to capture a wider spectrum of *cis*-regulatory activity.

As in previous versions of the registry[9](/articles/s41586-025-09909-9#ref-CR9 "ENCODE Project Consortium et al. Expanded encyclopaedias of DNA elements in the human and mouse genomes. Nature 583, 699–710 (2020)."), we performed cCRE classification across all biosamples (cell type-agnostic) and in specific biosamples (Supplementary Note [1.4](/articles/s41586-025-09909-9#MOESM1)). To facilitate comparisons at the organ and tissue level, we also generated aggregate annotations by combining biosamples from the same organ or tissue (Supplementary Note [1.5](/articles/s41586-025-09909-9#MOESM1) and Supplementary Table [4a](/articles/s41586-025-09909-9#MOESM3)).

This expanded classification scheme and inclusion of additional biosamples added 1.4 million human and 587,000 mouse cCREs, which were enriched for evolutionary conservation and regulatory activity (Supplementary Notes [1.6](/articles/s41586-025-09909-9#MOESM1) and [1.7](/articles/s41586-025-09909-9#MOESM1), Supplementary Figs. [4](/articles/s41586-025-09909-9#MOESM1) and [5](/articles/s41586-025-09909-9#MOESM1) and Supplementary Tables [4b](/articles/s41586-025-09909-9#MOESM3) and [5a,b](/articles/s41586-025-09909-9#MOESM3)). To evaluate the effect of this expansion, we compared the ENCODE4 cCREs to four datasets that recently annotated CREs using independent approaches across diverse tissues (Supplementary Note [1.7](/articles/s41586-025-09909-9#MOESM1)). For all four datasets, the fraction of overlapping ENCODE4 cCREs increased relative to ENCODE3 cCREs, driven by improved brain, embryonic tissue and immune coverage (Supplementary Fig. [6](/articles/s41586-025-09909-9#MOESM1) and Supplementary Table [5c–j](/articles/s41586-025-09909-9#MOESM3)). These

[... truncated ...]