---
url: https://t.co/fLQgUqvNNC
title: "Temporal tissue dynamics from a spatial snapshot | Nature"
clipped: 2026-01-22 12:28
source: browser-history
---

# Temporal tissue dynamics from a spatial snapshot | Nature

> Source: [https://t.co/fLQgUqvNNC](https://t.co/fLQgUqvNNC)

Temporal tissue dynamics from a spatial snapshot

[Download PDF](/articles/s41586-025-09876-1.pdf)

[Download PDF](/articles/s41586-025-09876-1.pdf)

### Subjects

* [Cancer models](/subjects/cancer-models)
* [Computational biophysics](/subjects/computational-biophysics)
* [Dynamical systems](/subjects/dynamical-systems)
* [Multicellular systems](/subjects/multicellular-systems)
* [Stochastic modelling](/subjects/stochastic-modelling)

## Abstract

Physiological and pathological processes such as inflammation and cancer emerge from interactions between cells over time[1](/articles/s41586-025-09876-1#ref-CR1 "Adler, M., Chavan, A. R. & Medzhitov, R. Tissue biology: in search of a new paradigm. Annu. Rev. Cell Dev. Biol. 39, 67–89 (2023)."). However, methods to follow cell populations over time within the native context of a human tissue are lacking because a biopsy offers only a single snapshot. Here we present one-shot tissue dynamics reconstruction (OSDR), an approach to estimate a dynamical model of cell populations based on a single tissue sample. OSDR uses spatial proteomics to learn how the composition of cellular neighbourhoods influences division rate, providing a dynamical model of cell population change over time. We apply OSDR to human breast cancer data[2](#ref-CR2 "Danenberg, E. et al. Breast tumor microenvironment structures are associated with genomic features and clinical outcome. Nat. Genet. 54, 660–669 (2022)."),[3](#ref-CR3 "Wang, X. Q. et al. Spatial predictors of immunotherapy response in triple-negative breast cancer. Nature 621, 868–876 (2023)."),[4](/articles/s41586-025-09876-1#ref-CR4 "Fischer, J. R. et al. Multiplex imaging of breast cancer lymph node metastases identifies prognostic single-cell populations independent of clinical classifiers. Cell Rep. Med. 4, 100977 (2023)."), and reconstruct two fixed points of fibroblasts and macrophage interactions[5](/articles/s41586-025-09876-1#ref-CR5 "Adler, M. et al. Principles of cell circuits for tissue repair and fibrosis. iScience 23, 100841 (2020)."),[6](/articles/s41586-025-09876-1#ref-CR6 "Zhou, X. et al. Circuit design features of a stable two-cell system. Cell 172, 744–757.e17 (2018)."). These fixed points correspond to hot and cold fibrosis[7](/articles/s41586-025-09876-1#ref-CR7 "Miyara, S. et al. Cold and hot fibrosis define clinically distinct cardiac pathologies. Cell Syst. 16, 101198 (2025)."), in agreement with co-culture experiments that measured these dynamics directly[8](/articles/s41586-025-09876-1#ref-CR8 "Mayer, S. et al. The tumor microenvironment shows a hierarchy of cell-cell interactions dominated by fibroblasts. Nat. Commun. 14, 5810 (2023)."). We then use OSDR to discover a pulse-generating excitable circuit of T and B cells in the tumour microenvironment, suggesting temporal flares of anticancer immune responses. Finally, we study longitudinal biopsies from a triple-negative breast cancer clinical trial[3](/articles/s41586-025-09876-1#ref-CR3 "Wang, X. Q. et al. Spatial predictors of immunotherapy response in triple-negative breast cancer. Nature 621, 868–876 (2023)."), in which OSDR predicts the collapse of the tumour cell population in responders but not in non-responders, based on early-treatment biopsies. OSDR can be applied to a wide range of spatial proteomics assays to enable analysis of tissue dynamics based on patient biopsies.

## Main

Physiological processes involve cell populations that change over time[1](/articles/s41586-025-09876-1#ref-CR1 "Adler, M., Chavan, A. R. & Medzhitov, R. Tissue biology: in search of a new paradigm. Annu. Rev. Cell Dev. Biol. 39, 67–89 (2023)."). Some cell populations expand and others are removed, as occurs in development and in the immune response to pathogens[9](/articles/s41586-025-09876-1#ref-CR9 "Medzhitov, R. The spectrum of inflammatory responses. Science 
                https://doi.org/10.1126/science.abi5200
                
               (2021)."). A clinically important example of changing cell populations is the evolution of the cancer microenvironment, in which the growing tumour recruits stromal and immune cells that are crucial for tumour survival[10](/articles/s41586-025-09876-1#ref-CR10 "Tlsty, T. D. & Coussens, L. M. Tumor stroma and regulation of cancer development. Annu. Rev. Pathol. 1, 119–150 (2006)."),[11](/articles/s41586-025-09876-1#ref-CR11 "De Visser, K. E. & Joyce, J. A. The evolving tumor microenvironment: from cancer initiation to metastatic outgrowth. Cancer Cell 41, 374–403 (2023)."). Understanding cell population dynamics and the underlying cell–cell communication circuits is a major goal of tissue biology, and can enable new treatment strategies based on sculpting cell populations in desired ways[7](/articles/s41586-025-09876-1#ref-CR7 "Miyara, S. et al. Cold and hot fibrosis define clinically distinct cardiac pathologies. Cell Syst. 16, 101198 (2025)."),[12](/articles/s41586-025-09876-1#ref-CR12 "Wang, S. et al. An autocrine signaling circuit in hepatic stellate cells underlies advanced fibrosis in nonalcoholic steatohepatitis. Sci. Transl. Med. 15, eadd3949 (2023).").

However, measuring the dynamics of cell populations in human tissues is currently very difficult. Biopsies provide a single snapshot, and taking multiple biopsies from a single patient is infeasible and does not provide longitudinal evidence from the same cells.

Current strategies to measure tissue dynamics do not apply to human biopsies. For example, cell lines, mice, organ-on-a-chip[13](/articles/s41586-025-09876-1#ref-CR13 "Leung, C. M. et al. A guide to the organ-on-a-chip. Nat. Rev. Methods Primer 2, 33 (2022).") or organoid models[14](/articles/s41586-025-09876-1#ref-CR14 "Kim, J., Koo, B.-K. & Knoblich, J. A. Human organoids: model systems for human biology and medicine. Nat. Rev. Mol. Cell Biol. 21, 571–584 (2020).") are treated as replicas and are analysed at different time points. Ex vivo tissues can be followed over time but lack the native physiological context. Intravital fluorescence microscopy has made it possible to follow living cells within animal models[15](/articles/s41586-025-09876-1#ref-CR15 "Pittet, M. J., Garris, C. S., Arlauckas, S. P. & Weissleder, R. Recording the wild lives of immune cells. Sci. Immunol. 3, eaaq0491 (2018)."),[16](/articles/s41586-025-09876-1#ref-CR16 "Entenberg, D., Oktay, M. H. & Condeelis, J. S. Intravital imaging to study cancer progression and metastasis. Nat. Rev. Cancer 23, 25–42 (2023)."). Recent advances use synthetic biology to engineer cells to record their activity or lineage[17](/articles/s41586-025-09876-1#ref-CR17 "Quinn, J. J. et al. Single-cell lineages reveal the rates, routes, and drivers of metastasis in cancer xenografts. Science 371, eabc1944 (2021)."),[18](/articles/s41586-025-09876-1#ref-CR18 "Horns, F. et al. Engineering RNA export for measurement and manipulation of living cells. Cell 
                https://doi.org/10.1016/j.cell.2023.06.013
                
               (2023)."). These approaches are restricted to animal models or in vitro settings, limiting their ability to capture the complexity of in vivo dynamics in humans.

The emergence of single-cell technologies offers new opportunities for studying tissues at high resolution. Notable approaches use single-cell data for understanding cell–cell communication at a single time point[19](#ref-CR19 "Efremova, M., Vento-Tormo, M., Teichmann, S. A. & Vento-Tormo, R. CellPhoneDB: inferring cell–cell communication from combined expression of multi-subunit ligand–receptor complexes. Nat. Protoc. 15, 1484–1506 (2020)."),[20](#ref-CR20 "Jerby-Arnon, L. & Regev, A. DIALOGUE maps multicellular programs in tissue from single-cell or spatial transcriptomics data. Nat. Biotechnol. 40, 1467–1477 (2022)."),[21](#ref-CR21 "Browaeys, R., Saelens, W. & Saeys, Y. NicheNet: modeling intercellular communication by linking ligands to target genes. Nat. Methods 17, 159–162 (2020)."),[22](/articles/s41586-025-09876-1#ref-CR22 "Armingol, E., Baghdassarian, H. M. & Lewis, N. E. The diversification of methods for studying cell–cell interactions and communication. Nat. Rev. Genet. 
                https://doi.org/10.1038/s41576-023-00685-8
                
               (2024)."). Other approaches attempt to infer dynamics of processes such as transcription within individual cells on a timescale of hours. Examples include RNA velocity, ergodic rate analysis and Zman-seq[23](#ref-CR23 "La Manno, G. et al. RNA velocity of single cells. Nature 560, 494–498 (2018)."),[24](#ref-CR24 "Kafri, R. et al. Dynamics extracted from fixed cells reveal feedback linking cell growth to cell cycle. Nature 494, 480–483 (2013)."),[25](/articles/s41586-025-09876-1#ref-CR25 "Kirschenbaum, D. et al. Time-resolved single-cell transcriptomics defines immune trajectories in glioblastoma. Cell 
                https://doi.org/10.1016/j.cell.2023.11.032
                
               (2023)."). These methods do not address the challenge of understanding how cell populations change on the tissue level, processes that could take days to weeks.

Here we present an approach to estimate cell population dynamics based on a spatial biopsy snapshot (Fig. [1a](/articles/s41586-025-09876-1#Fig1)). This approach, OSDR, is based on using a cell division marker to determine division rates as a function of neighbourhood composition, providing a dynamical model of how cell populations change over time. We applied OSDR to human breast cancer spatial proteomics samples from three large cohorts[2](#ref-CR2 "Danenberg, E. et al. Breast tumor microenvironment structures are associated with genomic features and clinical outcome. Nat. Genet. 54, 660–669 (2022)."),[3](#ref-CR3 "Wang, X. Q. et al. Spatial predictors of immunotherapy response in triple-negative breast cancer. Nature 621, 868–876 (2023)."),[4](/articles/s41586-025-09876-1#ref-CR4 "Fischer, J. R. et al. Multiplex imaging of breast cancer lymph node metastases identifies prognostic single-cell populations independent of clinical classifiers. Cell Rep. Med. 4, 100977 (2023)."). OSDR reconstructs a fibroblast–macrophage circuit with two steady states of hot and cold fibrosis[5](/articles/s41586-025-09876-1#ref-CR5 "Adler, M. et al. Principles of cell circuits for tissue repair and fibrosis. iScience 23, 100841 (2020)."),[7](/articles/s41586-025-09876-1#ref-CR7 "Miyara, S. et al. Cold and hot fibrosis define clinically distinct cardiac pathologies. Cell Syst. 16, 101198 (2025)."), in agreement with co-culture experiments that measure dynamics directly[8](/articles/s41586-025-09876-1#ref-CR8 "Mayer, S. et al. The tumor microenvironment shows a hierarchy of cell-cell interactions dominated by fibroblasts. Nat. Commun. 14, 5810 (2023)."). We then used OSDR to discover a pulse-generating excitable circuit of T and B cells, suggesting that cancer surveillance in the tumour microenvironment operates in temporal flares, as opposed to the steady-state picture implicit in current literature. Finally, we validated OSDR using longitudinal data from patients with triple-negative breast cancer who received either chemotherapy or chemotherapy and immunotherapy. In both treatment regimes, OSDR predicted the collapse of the tumour cell population in responders but not in non-responders, based on biopsies at treatment initiation. The present approach opens the way to infer cell population dynamics from spatial snapshots of patient biopsies.

**Fig. 1: Overview of OSDR.**

[![figure 1](//media.springernature.com/lw685/springer-static/image/art%3A10.1038%2Fs41586-025-09876-1/MediaObjects/41586_2025_9876_Fig1_HTML.png)](/articles/s41586-025-09876-1/figures/1)

**a**, OSDR uses spatial proteomics of a biopsy to infer cell population dynamics. The illustration was created by Nigel Orme. **b**, The division probability is learned based on the number of cells of each type in the neighbourhood. **c**, Cell division rates are inferred from a cell division marker (Ki67). **d**, Rate of change in the number of cells \(X\) is the difference between divisions and removals as described by a stochastic model. **e**, Simulations can take an initial spatial arrangement of cells and propagate the populations over time. **f**, OSDR also provides a phase portrait showing the direction of change of cell populations as a function of their neighbourhoods, revealing fixed points.

[Full size image](/articles/s41586-025-09876-1/figures/1)

## Tissue dynamics from a spatial snapshot

The aim of this study was to introduce a method for inferring a dynamical model of cell populations based on a single tissue sample (Fig. [1a](/articles/s41586-025-09876-1#Fig1)). We considered biopsy sections with cell-type markers and cell division markers. From this data, we generated a list of cell coordinates and division status for each cell type. We used imaging mass cytometry (IMC) samples[2](#ref-CR2 "Danenberg, E. et al. Breast tumor microenvironment structures are associated with genomic features and clinical outcome. Nat. Genet. 54, 660–669 (2022)."),[3](#ref-CR3 "Wang, X. Q. et al. Spatial predictors of immunotherapy response in triple-negative breast cancer. Nature 621, 868–876 (2023)."),[4](/articles/s41586-025-09876-1#ref-CR4 "Fischer, J. R. et al. Multiplex imaging of breast cancer lymph node metastases identifies prognostic single-cell populations independent of clinical classifiers. Cell Rep. Med. 4, 100977 (2023).") and the division marker Ki67 ([Methods](/articles/s41586-025-09876-1#Sec8)), but in principle, other spatial proteomic or transcriptomic assays can be used ([Methods](/articles/s41586-025-09876-1#Sec8)).

The key idea is to use a spatial omics snapshot to estimate the division and removal rates of cells as a function of the composition of the neighbourhood of the cell. We call this approach OSDR. Its essential advance is the use of division rates measured at the level of single cells at one time point to produce dynamics at the tissue level.

The change in cell counts within a tissue is a balance between division, death and migration. We separated the dynamics into two parts: one part that results from cell division and death within a tissue, and a second part that results from influx of cells from circulation. Our work focused on inferring cell division and death. We then used this estimated component to quantify the net contribution of migration from external sources of cells ([Methods](/articles/s41586-025-09876-1#Sec8); Supplementary Figs. [3m,n](/articles/s41586-025-09876-1#MOESM1) and [4m,n](/articles/s41586-025-09876-1#MOESM1)). In our analyses, local proliferation and removal were sufficient to explain the cell population dynamics, but this might not be the case in settings with a massive influx of cells.

To estimate the effect of each cell type on the growth and removal of other cells, we assumed that the neighbourhood surrounding each cell contains growth factors secreted by ne

[... truncated ...]