---
title: Inferring gene-regulatory networks using epigenomic priors
authors:
- Bartlett, T. E.
- Li, M.
- Song, C.
- Gao, Y.
- Huang, Q.
date: '2025-12-26'
categories:
- genomics
pdf_url: https://www.biorxiv.org/content/10.1101/2024.04.23.590858v7.full.pdf
paper_id: biorxiv_10.1101_2024.04.23.590858
source: biorxiv
tags:
- paper
- source/biorxiv
- topic/genomics
---

# Inferring gene-regulatory networks using epigenomic priors

**Authors:** Bartlett, T. E., Li, M., Song, C., Gao, Y., Huang, Q.

**Date:** 2025-12-26 | **Source:** biorxiv | **Categories:** genomics

[PDF](https://www.biorxiv.org/content/10.1101/2024.04.23.590858v7.full.pdf)

## Abstract

We show improved accuracy in-silico of inference of gene-regulatory network (GRN) structure, resulting from the use of an epigenomic prior network. We demonstrate important use-cases of our proposed methodology by re-analysing 12 datasets from 12 different studies. These include data from cells from human embryos, healthy adult tissue, and cancer, and include single-cell mRNA sequencing data, DNA methylation (DNAme) data, chromatin accessibility data, and histone modification data. We find that DNAme data are very effective for inferring the epigenomic prior network, recapitulating known epigenomic network structure found previously from chromatin accessibility data. Furthermore, we find that inferring the epigenomic prior network from DNAme data reveals candidate TF cis-regulations for at least eight times as many genes, when compared with chromatin accessibility data. When our proposed methodology is applied to real datasets from human embryonic development and from women at risk of breast cancer, we find patterns of differential cis-regulation that are in line with expectations under appropriate biological models, and that may be used to propose hypotheses about pre-cancerous epigenomic changes.

## Notes

